tools/src/main/java/org/apache/kafka/tools/SchemaChecker/MetadataSchemaChecker.java

```
@@ -0,0 +1,347 @@
+            "UnfenceBrokerRecord.json", "UnregisterBrokerRecord.json", "UserScramCredentialRecord.json", "ZkMigrationRecord.json"};
+    public static void main(String[] args) throws Exception {
+
+        try {
+            List<String> localContent = new ArrayList<>();
+            for(String jsonSchema: filesCheckMetadata) {
+                final String dir = System.getProperty("user.dir");
+                String path = dir + "/metadata/src/main/resources/common/metadata/" + jsonSchema;
+                BufferedReader reader = new BufferedReader(new FileReader(path));
+                for (int i = 0; i < 15; i++) {
+                    reader.readLine();
+                }
+                StringBuilder content = new StringBuilder();
+                for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+                    content.append(line);
+                }
+                localContent.add(content.toString());
+            }
+
+            List<String> gitContent = GetDataFromGit();
+            if (localContent.size() != gitContent.size()) {
+                throw new IllegalStateException("missing schemas");
+            }
+            for(int i = 0; i < localContent.size(); i++) {
+                if (Objects.equals(localContent.get(i), gitContent.get(i))) {
+                    continue;
+                }
+
+                ObjectMapper objectMapper = new ObjectMapper();
+                JsonNode jsonNode1 = objectMapper.readTree(gitContent.get(i));
+                JsonNode jsonNode2 = objectMapper.readTree(localContent.get(i));
+
+                checkApiTypeVersions(jsonNode1, jsonNode2);
+                parser((ArrayNode) jsonNode1.get("fields"), (ArrayNode) jsonNode2.get("fields"));
+            }
+
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred.");
```

I don't see why we have to catch this stack trace. Just let it propagate to the top level and then you can see the problem when running this on the command-line.

If you are going to catch it, you at least need to set the exit status code to be nonzero.